alibi/explainers/tests/test_simiarlity/test_grad_methods_integration.py

```diff
@@ -138,6 +139,7 @@ def test_correct_grad_dot_sim_result_tf(seed, normed_ds):
     `tensorflow` backend.
     """
     model = keras.Sequential([keras.layers.Dense(1, use_bias=False)])
+    model.build((None, 2))
```

Comment:
What was the reason for adding these? Since we're hard-coding the input dimension, probably should mention it in a comment or something?

Reply:
A newly instantialized model is built either explicitly as above or when the user passes input to it. Previously this was done in the grad method but I've since added these warnings that require first checking whether the model weights are trainable or not. If the model hasn't been built by this point an error is thrown. 

I figure the above behaviour is alright as users should only be passing build models to the method and if they aren't the error is descriptive enough for them to figure it out without us adding anything. I'll add a comment explaining it in the tests though.